src/main/java/io/cryostat/core/net/JFRConnection.java

```diff
@@ -146,6 +157,38 @@ public synchronized int getPort() {
         }
     }
 
+    public synchronized int getJvmId() {
+        try {
+            ObjectName runtimeBean = new ObjectName("java.lang:type=Runtime");
+            List<String> attributes =
+                    new ArrayList<>(
+                            Arrays.asList(
+                                    "ClassPath",
+                                    "Name",
+                                    "InputArguments",
+                                    "LibraryPath",
+                                    "VmVendor",
+                                    "VmVersion",
+                                    "StartTime"));
+            HashCodeBuilder jvmId = new HashCodeBuilder();
+
+            for (String attr : attributes) {
+                jvmId.append(
+                        this.rjmxConnection.getAttributeValue(
+                                new MRI(Type.ATTRIBUTE, runtimeBean, attr)));
+            }
+            return jvmId.hashCode();
+        } catch (MalformedObjectNameException
+                | IOException
+                | AttributeNotFoundException
+                | InstanceNotFoundException
+                | MBeanException
+                | ReflectionException e) {
+            cw.println(e);
```

Comment:
Where does this get printed to? 
I'm still looking for the cause of the NPE that sometimes occurs when I start an automated rule or perform a graphql query.

Reply:
Okay, thanks